src/transformers/generation_stopping_criteria.py

```diff
@@ -107,6 +107,23 @@ def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor, **kwa
         return time.time() - self.initial_timestamp > self.max_time
 
 
+class EndOfStringCriteria(StoppingCriteria):
+    """
+    This class can be used to stop generation whenever the eos_token_id is generated. 
+
+    Args:
+        eos_token_id (`int`):
+            The token id for the end of string (eos) token.
+    """
+
+    def __init__(self, eos_token_id: int):
+        self.eos_token_id = eos_token_id
+
+    @add_start_docstrings(STOPPING_CRITERIA_INPUTS_DOCSTRING)
+    def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor, **kwargs) -> bool:
+        return sum([self.eos_token_id in i for i in input_ids]) == input_ids.shape[0]
```

Comment:
I think this line is wrong and the following one is correct, no ?

Reply:
Hey @Narsil thanks for the helpful comments. And apologies for the few quality errors I was planning on addressing those after we decided on the strategy for eos_token_id. 

Here specifically I had 2 different returns because I was experimenting with different approaches to stop a beam search. One stops if any of the beams reach the eos_token_id the other waits for all of them to reach the eos_token_id. I'll remove both for now and address adding eos_token_id as a stopping criteria in a separate PR as per @gante's comments.

Reply:
Oh I see. I think beam search should wait on all beams being "done". (And you keep the eos ones has long as their score allows.

I didn't touch enough the beam search code to be sure how to handle that.
Also I think you only need to check then final tokens to prevent looping over the whole `input_ids`.

`input_ids[:, -1] == eos_token_id` no ?

Reply:
It's a bit more complex than that -- we can only check new tokens, but different batch sequences may generate `eos_token_id` at different times.

The generate functions use an auxiliary variable to keep track of which members of the batch have already finished, see [here](https://github.com/huggingface/transformers/blob/main/src/transformers/generation_utils.py#L1744)